classification/train.py

```diff
@@ -174,32 +180,37 @@ def valid(model: Any,
     label = []
 
     # tqdm is a progress bar
-    dl = tqdm(data_loader, position=5)
+    dl = tqdm(data_loader, position=5, total=int(len(data_loader)*dataset_ratio))
 
     if CONFIG.map_debug and CONFIG.model_checkpoint is not None:
         pred = torch.load("/".join(CONFIG.model_checkpoint.split('/')[:-1]) + '/pred.pt')
         label = torch.load("/".join(CONFIG.model_checkpoint.split('/')[:-1]) + '/label.pt')
     else:
-        for _, (mels, labels) in enumerate(dl):
-            mels = mels.to(device)
-            labels = labels.to(device)
-
-            # argmax
-            outputs = model(mels)
-            check_shape(outputs, labels)
-
-            loss = model.loss_fn(outputs, labels)
-
-            running_loss += loss.item()
-
-            pred.append(outputs.cpu().detach())
-            label.append(labels.cpu().detach())
-
-        pred = torch.cat(pred)
-        label = torch.cat(label)
-        if CONFIG.map_debug and CONFIG.model_checkpoint is not None:
-            torch.save(pred, "/".join(CONFIG.model_checkpoint.split('/')[:-1]) + '/pred.pt')
-            torch.save(label, "/".join(CONFIG.model_checkpoint.split('/')[:-1]) + '/label.pt')
+        with torch.no_grad():
+            for index, (mels, labels) in enumerate(dl):
```

Comment:
Double check to make sure validation dl is random shuffle

Reply:
We good